npm/qsharp/markdown_latex_plugin.js

```diff
@@ -0,0 +1,273 @@
+// @ts-check
+/// <reference lib="es2022"/>
+
+/***** LaTeX passthrough plug-in for the markdown-it parser *****
+
+When converting Markdown to HTML that may contain LaTeX, but wanting to
+leave the LaTeX untouched, there are a number of hazards. These include:
+
+- Markdown escapes in the LaTeX, such as double escapes ("\\" and lines
+  ending with a "\"), will be processed, corrupting the LaTeX. This sequence
+  is common in \begin{bmatrix} structures (which is common in Quantum).
+
+- The LaTeX may be processed for Markdown sequences. Common examples
+  here include finding two "_" chars in the LaTeX and converting this to
+  emphasis "<em>" tags around the content.
+
+Disabling Markdown escapes can avoid the former, but not the latter, which
+requires re-writing the LaTeX if possible to avoid, and can be subtle to catch.
+
+To avoid these issues, this plug-in detects $..$ and $$..$$ content in
+raw Markdown and inline HTML, and passes it through the Markdown parser as-is.
+
+*/
+
+// To help make type checking annotations cleaner
+/** @typedef {import("markdown-it/dist/markdown-it.js")} MarkdownIt */
+/** @typedef {import("markdown-it/dist/markdown-it.js").StateInline} StateInline */
+
+// Below code to locate LaTeX blocks largely taken from @vscode/markdown-it-katex
```

Comment:
As outlined, this code is largely a portion of the existing KaTeX plugin, it just passes the LaTeX through instead of calling KaTeX on it.

Reply:
A GH link here to the source would be nice:
https://github.com/microsoft/vscode-markdown-it-katex/blob/9f3e1dff0fa2e011c63cb6a05fa6e80b7624538f/src/index.ts

Mainly because it helps me avoid reviewing the parser code, since I can confirm it's a direct copy and paste ;)

Reply:
Done.